crates/networking/rpc/engine/fork_choice.rs

```diff
@@ -211,63 +211,54 @@ async fn handle_forkchoice(
         .await?
     {
-        warn!(
-            "Invalid fork choice state. Reason: Invalid ancestor {:#x}",
-            latest_valid_hash
-        );
-        Err(InvalidForkChoice::InvalidAncestor(latest_valid_hash))
-    } else {
-        // Check parent block hash in invalid_ancestors (if head block exists)
-        let head_block = context
-            .storage
-            .get_block_header_by_hash(fork_choice_state.head_block_hash)?;
-        let check_parent = if let Some(head_block) = head_block {
-            debug!(
-                "Checking parent for invalid ancestor {}",
-                head_block.parent_hash
-            );
-            context
-                .storage
-                .get_latest_valid_ancestor(head_block.parent_hash)
-                .await
-                .ok()
-                .flatten()
-        } else {
-            None
-        };
+        return Ok((
+            None,
+            ForkChoiceResponse::from(PayloadStatus::invalid_with(
+                latest_valid_hash,
+                InvalidForkChoice::InvalidAncestor(latest_valid_hash).to_string(),
+            )),
+        ));
+    }
 
-        // Check head block hash in invalid_ancestors
-        if let Some(latest_valid_hash) = check_parent {
-            Err(InvalidForkChoice::InvalidAncestor(latest_valid_hash))
-        } else {
-            // Check if there is an ongoing sync before applying the forkchoice
-            match context.syncer.status()? {
-                // Apply current fork choice
-                SyncStatus::Inactive => {
-                    // All checks passed, apply fork choice
-                    apply_fork_choice(
-                        &context.storage,
-                        fork_choice_state.head_block_hash,
-                        fork_choice_state.safe_block_hash,
-                        fork_choice_state.finalized_block_hash,
-                    )
-                    .await
-                }
-                // Restart sync if needed
-                _ => Err(InvalidForkChoice::Syncing),
-            }
+    // Check parent block hash in invalid_ancestors (if head block exists)
+    if let Some(head_block) = context
+        .storage
+        .get_block_header_by_hash(fork_choice_state.head_block_hash)?
+    {
+        if let Some(latest_valid_hash) = context
+            .storage
+            .get_latest_valid_ancestor(head_block.parent_hash)
+            .await?
+        {
+            return Ok((
+                None,
+                ForkChoiceResponse::from(PayloadStatus::invalid_with(
+                    latest_valid_hash,
+                    InvalidForkChoice::InvalidAncestor(latest_valid_hash).to_string(),
+                )),
+            ));
         }
-    };
+    }
 
-    match fork_choice_res {
+    if context.syncer.sync_mode() == SyncMode::Snap {
+        warn!("Snap sync in progress, setting new head optimistically");
```

Comment:
This sounds misleading, `setting new head optimistically` sound like we are setting it as canonical head of the chain. Also `warn` is a bit too harsh for something that is regular behaviour. It is also too spammy as fcu happen quite often

Reply:
Agreed with `warn`, but aren't we setting the new head as the canonical head of the chain in this case?

Reply:
Removed the warning